### PR TITLE
Feature/issue 21

### PR DESCRIPTION
**Pull request by ovaistariq**

### CODE REVIEW

**Comment by ovaistariq** [comment]
I have created this PR so that it can be reviewed. But the download part still needs to be fixed

File: tests/integration/conftest.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
+import pytest
+import os
+
+from subprocess import call
+from twindb_backup.destination.s3 import S3
+
+BUCKET = 'twindb-backup-test-travis-%s' % os.environ['TRAVIS_JOB_NUMBER']
```
Comment (akuzminsky) [inline tests/integration/conftest.py]: if we run the integration test locally `TRAVIS_JOB_NUMBER` won't exist. Should we handle it and replace with a random number or something?

File: twindb_backup/destination/s3.py [hunk 2]
```diff
@@ -131,3 +299,30 @@ def _status_exists(self):
             else:
                 raise
         return False
+
+    @staticmethod
+    def validate_client_response(response):
+        """Validates the response returned by the client. Raises an exception
+            if the response code is not 200
```
Comment (akuzminsky) [inline twindb_backup/destination/s3.py]: ... or 204

**Comment by akuzminsky** [comment]
I wonder if the PR closes #3

**Pull request merged**